pkg/client/clientcmd/client_config_test.go

```diff
@@ -141,11 +141,49 @@ func TestCreateClean(t *testing.T) {
 	}
 
 	matchStringArg(config.Clusters["clean"].Server, clientConfig.Host, t)
+	matchStringArg("", clientConfig.Prefix, t)
 	matchStringArg(config.Clusters["clean"].APIVersion, clientConfig.Version, t)
 	matchBoolArg(config.Clusters["clean"].InsecureSkipTLSVerify, clientConfig.Insecure, t)
 	matchStringArg(config.AuthInfos["clean"].Token, clientConfig.BearerToken, t)
 }
 
+func TestCreateCleanWithPrefix(t *testing.T) {
+	tt := []struct {
+		server string
+		host   string
+		prefix string
+	}{
+		{"https://anything.com:8080/foo/bar", "https://anything.com:8080", "/foo/bar"},
+		{"http://anything.com:8080/foo/bar", "http://anything.com:8080", "/foo/bar"},
+		{"http://anything.com:8080/foo/bar/", "http://anything.com:8080", "/foo/bar/"},
+		{"http://anything.com:8080/", "http://anything.com:8080/", ""},
+		{"http://anything.com:8080//", "http://anything.com:8080", "//"},
+		{"anything.com:8080/foo/bar", "anything.com:8080/foo/bar", ""},
```

Comment:
Yes, empty because the parse fails. I haven't tested with a kubeconfig file
-- can you point me to a kubeclient config test that I can play with?

On Thu, Jun 4, 2015 at 12:03 PM, David Eads <notifications@github.com>
wrote:

> In pkg/client/clientcmd/client_config_test.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9171#discussion_r31735536>
> :
>
> >  	matchStringArg(config.Clusters["clean"].APIVersion, clientConfig.Version, t)
> >  	matchBoolArg(config.Clusters["clean"].InsecureSkipTLSVerify, clientConfig.Insecure, t)
> >  	matchStringArg(config.AuthInfos["clean"].Token, clientConfig.BearerToken, t)
> >  }
> >
> > +func TestCreateCleanWithPrefix(t *testing.T) {
> > +	tt := []struct {
> > +		server string
> > +		host   string
> > +		prefix string
> > +	}{
> > +		{"https://anything.com:8080/foo/bar", "https://anything.com:8080", "/foo/bar"},
> > +		{"http://anything.com:8080/foo/bar", "http://anything.com:8080", "/foo/bar"},
> > +		{"http://anything.com:8080/foo/bar/", "http://anything.com:8080", "/foo/bar/"},
> > +		{"http://anything.com:8080/", "http://anything.com:8080/", ""},
> > +		{"http://anything.com:8080//", "http://anything.com:8080", "//"},
> > +		{"anything.com:8080/foo/bar", "anything.com:8080/foo/bar", ""},
>
> This is empty because the URL parse fails? What is the current behavior
> when this value is included in kubeconfig file?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9171/files#r31735536>
> .
>